syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package terminal;

/**
  终端服务
 */
service TerminalService {

  /**
     登录
   */
  rpc login (UserRequest) returns (stream Result) {}

  /**
     注册
   */
  rpc registerUser (UserRequest) returns (Result) {}

}

/**
  用户请求
 */
message UserRequest {
  string username = 1;
  string password = 2;
}

/**
  通用结果
 */
message Result {
  int64 code = 1;
  string message = 2;
}
